test/src/test/java/org/corfudb/utils/LockStoreTest.java

```diff
@@ -0,0 +1,283 @@
+                try {
+                    renewed1 = lockStore1.renew(lockId);
+                } catch (LockStoreException le) {
+                    renewed1 = false;
+                }
+            });
+            scheduleConcurrently(f -> {
+                try {
+                    renewed2 = lockStore2.renew(lockId);
+                } catch (LockStoreException le) {
+                    renewed2 = false;
+                }
+            });
+        } else {
+            scheduleConcurrently(f -> {
+                try {
+                    acquired1 = lockStore1.acquire(lockId);
+                } catch (LockStoreException le) {
+                    acquired1 = false;
+                }
+            });
+
+            scheduleConcurrently(f -> {
+                try {
+                    acquired2 = lockStore2.acquire(lockId);
+                } catch (LockStoreException le) {
+                    acquired2 = false;
+                }
+            });
+        }
+        executeScheduled(2, TIMEOUT, TimeUnit.SECONDS);
+    }
+
+    // Two runtimes try to acquire a lock with the same lock id concurrently.
+    // Verify that only one of them succeeds.
+    // Repeat this operation multiple times to expose race in this workflow,
+    // if any.  Each iteration clears the lock table and creates a new
+    // LockStore instance to exercise the init scenario.
+    @Test
+    public void testAcquireConcurrently() throws Exception {
+        for (int i=0; i<NUM_ITERATIONS; i++) {
+
+            LockStore lockStore1 = new LockStore(runtime1, UUID.randomUUID());
+            LockStore lockStore2 = new LockStore(runtime2, UUID.randomUUID());
+
+            executeAcquireOrRenewConcurrently(lockStore1, lockStore2, false);
+
+            Assert.assertTrue(acquired1 || acquired2);
+            Assert.assertFalse(acquired1 && acquired2);
+
+            cleanUp();
+        }
+    }
+
+    // Two runtimes try to acquire a lock with the same lock id concurrently.
+    // Verify that only one of them succeeds.
+    // Repeat this operation multiple times to expose race in this workflow,
+    // if any.
+    // Lock table is not cleared between iterations and the LockStore
+    // instance is reused.  This test simulates the case where acquire() is
+    // invoked even if there is a lock in the table(there is a lock owner).
+    @Test
+    public void testRepeatedAcquire() throws Exception {
+        Lock.setLeaseDuration(LOCK_LEASE_DURATION_LONG_SECONDS);
+
+        LockStore lockStore1 = new LockStore(runtime1, UUID.randomUUID());
+        LockStore lockStore2 = new LockStore(runtime2, UUID.randomUUID());
+
+        executeAcquireOrRenewConcurrently(lockStore1, lockStore2, false);
+
+        Assert.assertTrue(acquired1 || acquired2);
+        Assert.assertFalse(acquired1 && acquired2);
+
+        CorfuStore corfuStore1 = new CorfuStore(runtime1);
+        TxnContext txn = corfuStore1.txn(CORFU_SYSTEM_NAMESPACE);
```

Comment:
nit: Is it better to have all the txns in a try block?

Reply:
+1 you're leaking resources, it should be done in this way or explicitly closing the transactions.